src/main/java/xbot/common/subsystems/drive/SwerveKinematicsCalculator.java

```diff
@@ -0,0 +1,298 @@
+    final double endPosition;
+    final double acceleration;
+    final double initialVelocity;
+    final double goalVelocity;
+    final double maxVelocity;
+    final List<SwerveCalculatorNode> nodeMap;
+    RobotAssertionManager assertionManager;
+
+    final double totalOperationTime;
+    final double totalOperationDistance;
+
+    // Returns the x-intercepts of a quadratic equation
+    private static List<Double> quadraticFormula(double a, double b, double c) {
+        double squareRootResult = Math.sqrt(Math.pow(b, 2) - (4 * a * c));
+        double result1 = (-b + squareRootResult) / (2 * a);
+        double result2 = (-b - squareRootResult) / (2 * a);
+
+        List<Double> results = new ArrayList<>();
+        results.add(result1);
+        results.add(result2);
+        return results;
+    }
+
+    public static double calculateTimeToGoalPosition(double acceleration, double initialVelocity,
+                                                     double initialPosition, double goalPosition) {
+        double a = 0.5 * acceleration;
+        double b = initialVelocity;
+        double c = initialPosition - goalPosition;
+        if (acceleration == 0) {
+            return (-c/b);
+
+        } else if (acceleration > 0) {
+            List<Double> result = quadraticFormula(a, b, c);
+            return Math.max(result.get(0), result.get(1));
+
+        } else {
+            List<Double> result = quadraticFormula(-a, -b, -c);
+            return Math.min(result.get(0), result.get(1));
+
+        }
+    }
+
+    public SwerveKinematicsCalculator(RobotAssertionManager assertionManager, double startPosition, double endPosition,
+                                      SwervePointKinematics kinematics) {
+        this.assertionManager = assertionManager;
+        // Gimmicky way to avoid division by zero
+        if (endPosition - startPosition == 0) {
+            assertionManager.throwException("Calculator instantiated for same start and end position", new Exception());
+            endPosition += 0.01;
+        }
+
+        this.startPosition = startPosition;
+        this.endPosition = endPosition;
+        this.acceleration = kinematics.acceleration;
+        this.initialVelocity = kinematics.initialVelocity;
+        this.goalVelocity = kinematics.goalVelocity;
+        this.maxVelocity = kinematics.maxVelocity;
+
+        //TODO: Likely need to validate the above values to prevent bad stuff, works just fine serving for SwerveSimpleTrajectory though.
+        nodeMap = generateNodeMap();
+
+        double time = 0;
+        for (SwerveCalculatorNode node : this.nodeMap) {
+            time += node.operationTime();
+        }
+        totalOperationTime = time;
+        totalOperationDistance = Math.abs(endPosition - startPosition);
+    }
+
+    // Based off of given value, initialize the proper path, store the nodes in a list, and be prepared to output values
+    public List<SwerveCalculatorNode> generateNodeMap() {
```

Comment:
❓ is this something that ever needs to be called externally since it's called by the constructor?